### PR TITLE
Use python tokenizer to identify single-line comments and docstrings.

### PR DESCRIPTION
**Pull request by ashanbrown**
Perform the raw analysis using a single iterator.
Add test cases that cover line continuations and unusual docstrings.

### CODE REVIEW

**Comment by ashanbrown** [comment]
Ciao @rubik .  I was having trouble performing a raw analyze on a codebase so I rewrote much of the comment recognizing code from an early PR.   In particular, the code I was having a problem with looked like this:

```
def function():
    multiline_with_equals_in_it = """ """
    pass
```

There is some existing code that tries to recognize the position of '=' in multi-line string assignments that makes assumptions that may not always be the case.  I have replaced this code with code that uses the AST to recognize strings and comments rather than using a custom parser.  I've also tried to reduce the number of times radon parses the code to a single time.

I have only changed one existing test case, which appeared to be expecting the wrong number of blank lines.  I'm assuming we should count lines as blank whether or not they are inside multi-line strings.  

Please let me know what I can do to get this into the codebase.  Thank you.

File: radon/tests/test_raw.py [hunk 1]
```diff
@@ -187,7 +187,7 @@ def fib(n):
          """
          if n <= 1: return 1  # otherwise it will melt the cpu
          return fib(n - 2) + fib(n - 1)
-     ''', (6, 9, 10, 2, 3, 1, 1)),
+     ''', (6, 9, 10, 2, 3, 2, 1)),
```
Comment (ashanbrown) [inline radon/tests/test_raw.py]: I believe that '2' is the correct value for number of blank lines in this code.  I think previously the line in the multiline string had been excluded.

**Comment by rubik** [comment]
Thanks Andrew! I'll review this tomorrow but at a cursory glance it looks good. The code you provided definitely crashes the analysis, so something will have to be done.

**Comment by rubik** [comment]
Sorry for the delay Andrew! The PR looks good and I'm merging it. Thanks again!

**Comment by ashanbrown** [comment]
Thanks @rubik .  Please let me know if there are any reported issues with this change.  I'll be happy to try to fix them.

**Pull request merged**